six/two/two.cpp

```diff
@@ -370,24 +370,36 @@ PyObject *Two::_findSubclassOf(PyObject *base, PyObject *module)
 
     PyObject *dir = PyObject_Dir(module);
     if (dir == NULL) {
+        PyErr_Clear();
         setError("there was an error calling dir() on module object");
         return NULL;
     }
 
     PyObject *klass = NULL;
     for (int i = 0; i < PyList_GET_SIZE(dir); i++) {
-        // get symbol name
-        char *symbol_name;
+        // get the symbol in current list item
         PyObject *symbol = PyList_GetItem(dir, i);
-        if (symbol != NULL) {
-            symbol_name = PyString_AsString(symbol);
+        if (symbol == NULL) {
+            // This should never happen as it means we're out of bounds
+            PyErr_Clear();
+            setError("there was an error browsing dir() output");
+            goto done;
```

Comment:
nit: I believe we need to set `klass = NULL` before this, just in case it was set to a non-NULL value in a previous iteration of the loop. Or more simply, each iteration should both `DECREF` `klass` and set it to `NULL` before `continue`-ing?